### PR TITLE
VideoPress: add ClipboardButtonInput component

### LINKED ISSUE #25724: VideoPress Components: ClipboardButtonInput
Image

* Switch the button to Copied and to Copy again after, say, 3 seconds
* Select the entire text for reassurance and for people who want to still copy it manually (I could see someone saving it locally to a clipboard utility or something)

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds the ClipboardButtonInput component to the VideoPress package. If it's good enough, we can migrate it to Jetpack components.

image

Fixes https://github.com/Automattic/jetpack/issues/25724

#### Changes proposed in this Pull Request:

* VideoPress: add ClipboardButtonInput component

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Use storybook
* Play with the component
* The storybook contains doc about the component API

https://user-images.githubusercontent.com/77539/185691643-cdd5b29a-5bae-4ccf-98e6-4ab2bd73507d.mov

### CODE REVIEW

File: projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * Video Details Card component
+ *
+ * @param {ClipboardButtonInput} props - Component props.
+ * @returns {React.ReactNode} - VideoDetailsCard react component.
+ */
+const VideoDetailsCard: React.FC< ClipboardButtonInput > = ( {
+	text,
+	value,
+	onCopy = noop,
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx]: You don't need to use a `noop` function, now with optional chaining, you can use it.

```
onCopy?.();
```

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#syntax

File: projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	/* noop */
+};
+
+/**
+ * Video Details Card component
+ *
+ * @param {ClipboardButtonInput} props - Component props.
+ * @returns {React.ReactNode} - VideoDetailsCard react component.
+ */
+const VideoDetailsCard: React.FC< ClipboardButtonInput > = ( {
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx]: I started to see some points about not using `React.FC`, I think we can take a look on it after.

https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components
https://github.com/facebook/create-react-app/pull/8177

Comment (retrofox) [reply projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx]: Let me fix it in a follow-up, please

File: projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/admin/components/clipboard-button-input/index.tsx]: Awesome. Updated 🙇‍♂️

**Review by renatoagds — DISMISSED** [review]
LGTM

**Review by renatoagds — APPROVED** [review]

**Comment by retrofox** [comment]
r251067-wpcom

**Pull request merged**